scripts/download_modis.py

```diff
@@ -0,0 +1,165 @@
+        cloud_fraction_ice = (
+            daily_data.select("Cloud_Retrieval_Fraction_Ice").get().astype(np.int32)
+        )
+        cloud_fraction_ice_counts = (
+            daily_data.select("Cloud_Retrieval_Fraction_Ice_Pixel_Counts")
+            .get()
+            .astype(np.int32)
+        )
+        cloud_fraction = daily_data.select("Cloud_Fraction_Mean").get().astype(np.int32)
+        cloud_fraction_counts = (
+            daily_data.select("Cloud_Fraction_Pixel_Counts").get().astype(np.int32)
+        )
+        cloud_fraction_day = (
+            daily_data.select("Cloud_Fraction_Day_Mean").get().astype(np.int32)
+        )
+        cloud_fraction_day_counts = (
+            daily_data.select("Cloud_Fraction_Day_Pixel_Counts").get().astype(np.int32)
+        )
+
+        date_idx = date.timetuple().tm_yday - 1
+        unix_time = int((date - datetime(1970, 1, 1)).total_seconds())
+        dataset["date"][date_idx] = unix_time
+        dataset["Cloud_Water_Path_Ice_Mean"][date_idx, :, :] = cloud_water_path_ice_mean
+        dataset["Cloud_Water_Path_Ice_Histogram_Counts"][
+            date_idx, :, :, :
+        ] = cloud_water_path_ice_histo
+        dataset["Cloud_Retrieval_Fraction_Ice"][date_idx, :, :] = cloud_fraction_ice
+        dataset["Cloud_Retrieval_Fraction_Ice_Pixel_Counts"][
+            date_idx, :, :
+        ] = cloud_fraction_ice_counts
+        dataset["Cloud_Fraction_Mean"][date_idx, :, :] = cloud_fraction
+        dataset["Cloud_Fraction_Pixel_Counts"][
+            date_idx,
+            :,
+            :,
+        ] = cloud_fraction_counts
+        dataset["Cloud_Fraction_Day_Mean"][date_idx, :, :] = cloud_fraction_day
+        dataset["Cloud_Fraction_Day_Pixel_Counts"][
+            date_idx, :, :
+        ] = cloud_fraction_day_counts
+        dataset["date_idx"][date_idx] = date_idx
+
+    except Exception as e:
+        print(f"Error processing {downloaded_file_name}: {e}")
+
+
+start_date = datetime(year, 1, 1)  # start from Jan 1.
+bash_script_path = "./download_modis_daily.sh"
+destination_path = "./data_tmp/"
+
+# initialize NetCDF file
+nc_filename = f"ccic_modis_data_{year}.nc"
+dataset = nc.Dataset(nc_filename, "w", format="NETCDF4")
+
+dataset.createDimension("date_idx", None)
+date_idxs = dataset.createVariable("date_idx", "i4", ("date_idx",))
+
+dataset.createDimension("date", None)
+dates = dataset.createVariable("date", "i8", ("date",))
+dates.units = "Unix time [s]"
+
+dataset.createDimension("lat", 180)
+lats = dataset.createVariable("lat", "f4", ("lat",))
+lats[:] = np.arange(-89.5, 90.5, 1)
+dataset.createDimension("lon", 360)
+lons = dataset.createVariable("lon", "f4", ("lon",))
+lons[:] = np.arange(-179.5, 180.5, 1)
+
+dataset.createDimension("histo_bins", 16)
+histo_bins = dataset.createVariable(
+    "Cloud_Water_Path_Ice_Histo_Intervals", "i2", ("histo_bins",)
+)
+histo_bins[:] = np.array(
+    [5, 15, 35, 75, 125, 175, 225, 275, 325, 375, 425, 475, 750, 1500, 3000, 5000]
```

Comment:
Again in the [file spec](https://atmosphere-imager.gsfc.nasa.gov/sites/default/files/ModAtmo/MOD08_D3_fs_3040.txt), at the very end of the page, under 'Cloud_Water_Path_Ice_Histo_Intervals'.
I do admit that it would have been much better/clearer to extract this directly from the file!

Reply:
I think I see the bins that you saw in the ATBD, on page 116. I used the data set 08_D3 (level-3 daily means), and the bins listed in the ATBD are for the data set 06_L2 (level-2 data). I'm not sure why they use different bins, it seems strange. But I trust the file spec for the 08_D3 data has the correct bins for 08_D3.